src/OrleansProviders/Streams/Common/PooledCache/ChronologicalEvictionStrategy.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System;
+using System.Collections.Generic;
+using Orleans.Runtime;
+using System.Threading;
+
+namespace Orleans.Providers.Streams.Common
+{
+    /// <summary>
+    /// Eviction strategy that evicts data based off of age.
+    /// </summary>
+    /// <typeparam name="TCachedMessage"></typeparam>
+    public abstract class ChronologicalEvictionStrategy<TCachedMessage> : IEvictionStrategy<TCachedMessage>
+        where TCachedMessage : struct
```

Comment:
The shared eviction strategy by those three stream providers. It purge items based on Time purge predicate. And free blocks to the pool whenever it is purged, except for the current block in use